### PR TITLE
seqtest: migrate test-infra testify

### LINKED ISSUE #28076: migrate test-infra to testify for `executor/seqtest`

### PR DESCRIPTION
**Pull request by tisonkun**
Signed-off-by: tison <wander4096@gmail.com>

### What problem does this PR solve?

Issue Number: close #28076

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by mmyj** [comment]
I will review this in the weekend.

File: executor/seqtest/prepared_serial_test.go [hunk 1]
```diff
@@ -819,57 +856,59 @@ func (msm *mockSessionManager1) ShowProcessList() map[uint64]*util.ProcessInfo {
...
-	tk1 := testkit.NewTestKit(c, s.store)
+	tk1 := testkit.NewTestKit(t, store)
 
-	var err error
-	tk1.Se, err = session.CreateSession4Test(s.store)
-	c.Assert(err, IsNil)
-	tk1.GetConnectionID()
+	se, err := session.CreateSession4Test(store)
+	require.NoError(t, err)
+	tk1.SetSession(se)
```
Comment (mmyj) [inline executor/seqtest/prepared_serial_test.go:890]: The old `util/testkit.NewTestKit` will not create a session for testkit, but the new one `testkit.NewTestKit` will. Should we need to set again?

**Comment by tisonkun** [comment]
@mmyj thanks for your review! I'll take a look today.

File: executor/seqtest/prepared_serial_test.go (hunk 1 above)
Comment (tisonkun) [reply executor/seqtest/prepared_serial_test.go:890]: Removed.

File: testkit/testkit.go [hunk 2]
```diff
@@ -48,12 +49,23 @@ func NewTestKit(t *testing.T, store kv.Storage) *TestKit {
...
+		t:       t,
 		store:   store,
 		session: newSession(t, store),
 	}
 }
 
-// Session return a session
+// RefreshSession set a new session for the testkit
+func (tk *TestKit) RefreshSession() {
+	tk.session = newSession(tk.t, tk.store)
```
[the remark this answers is not in the document]
Comment (tisonkun) [reply testkit/testkit.go:60]: I tend to keep it as is:

1. It is the original implementation. Even we want to make change, I'd prefer to file another issue and investigate as it is not trivial.
2. Previous session can be referred outside and it's possible to be closed outside.
3. Dangling session won't leak goroutines.

File: testkit/testkit.go [hunk 3]
```diff
@@ -48,12 +49,23 @@ func NewTestKit(t *testing.T, store kv.Storage) *TestKit {
...
+}
+
+// SetSession set the session of testkit
+func (tk *TestKit) SetSession(session session.Session) {
+	tk.session = session
```
[the remark this answers is not in the document]
Comment (tisonkun) [reply testkit/testkit.go:65]: reply in https://github.com/pingcap/tidb/pull/28298#discussion_r717112692

**Review by mmyj — APPROVED** [review]
lgtm, and please resolve conflicting.

File: executor/seqtest/main_test.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+import (
+	"testing"
+
+	"github.com/pingcap/tidb/util/testbridge"
+	"github.com/tikv/client-go/v2/config"
+	"go.uber.org/goleak"
+)
+
+func TestMain(m *testing.M) {
```
Comment (XuHuaiyu) [inline executor/seqtest/main_test.go:25]: What does this function mean?

Comment (tisonkun) [reply executor/seqtest/main_test.go:25]: You can take a look at Golang test document about [`TestMain`](https://pkg.go.dev/testing#hdr-Main).

Generally it is used to do package level setup logic for tests. We use it here to inject goleak verification and other setups.

**Review by XuHuaiyu — APPROVED** [review]
LGTM

**Comment by tisonkun** [comment]
@XuHuaiyu @mmyj could you help on merging this PR？

**Pull request merged**